### PR TITLE
Add the hasDomainValues attribute to kore

### PR DESCRIPTION
**Pull request by virgil-serbanuta**

### CODE REVIEW

File: kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java [hunk 1]
```diff
@@ -114,6 +120,10 @@ public String convert(boolean heatCoolEq) {
         collectionSorts.add("MAP.Map");
         collectionSorts.add("LIST.List");
         collectionSorts.add("ARRAY.Array");
+        if (attributes.containsKey("token")) {
+            assert(!attributes.containsKey(HAS_DOMAIN_VALUES));
```
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java]: I'm reasonably sure this assertion could fail if the user added the attribute to their definition. In general it should be considered a bug for any k definition to generate an assertion failure when compiled

File: kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java [hunk 2]
```diff
@@ -137,6 +147,9 @@ public String convert(boolean heatCoolEq) {
                     sb.append("hooked-");
                 }
             }
+            if (tokenSorts.contains(sort)) {
+                att = att.add(HAS_DOMAIN_VALUES, "");
```
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java]: Don't think you need the second parameter here

**Review by dwightguth — CHANGES_REQUESTED** [review]
Looks good but a few things need to be improved

File: kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java (hunk 2 above)
Comment (virgil-serbanuta) [reply kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java]: Fixed.

File: kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java (hunk 1 above)
Comment (virgil-serbanuta) [reply kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java]: Fixed.

**Review by dwightguth — APPROVED** [review]

**Pull request merged**